src/attached_probe.cpp

```diff
@@ -329,16 +333,17 @@ std::string AttachedProbe::eventname() const
   }
 }
 
-static uint64_t resolve_offset(const std::string &path,
-                               const std::string &symbol,
-                               uint64_t loc)
+Result<uint64_t> AttachedProbe::resolve_offset(const std::string &path,
+                                               const std::string &symbol,
+                                               uint64_t loc)
 {
   bcc_symbol bcc_sym;
 
   if (bcc_resolve_symname(
-          path.c_str(), symbol.c_str(), loc, 0, nullptr, &bcc_sym))
-    throw util::FatalUserException("Could not resolve symbol: " + path + ":" +
+          path.c_str(), symbol.c_str(), loc, 0, nullptr, &bcc_sym)) {
+    return make_error<AttachError>("Could not resolve symbol: " + path + ":" +
                                    symbol);
```

Comment:
Aside: excellent place to use std::format and modernize the code in this file...

Reply:
If we don't remove std::format usage ;)

Reply:
Going to look at all the format conversion in another PR as soon as we decide what we want to do.